### PR TITLE
Document the OSM Importer Supported Keywords.

### PR DESCRIPTION
**Pull request by DavidMansolino**
It was not obvious for the user to know which of the OSM keywords the importer script is supporting.
https://www.cyberbotics.com/doc/automobile/openstreetmap-importer?version=document-OSM-importer-keyword#supported-osm-keywords

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:+1:

File: docs/automobile/openstreetmap-importer.md [hunk 1]
```diff
@@ -106,6 +103,96 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
 In addition to editing map from OpenStreetmap, JOSM is also very convenient to create new environment from scratch.
 You can see in the [previous picture](#left-the-osm-file-created-in-josm-right-the-resulting-world-open-in-webots-after-conversion) a map fully created in JOSM and then exported and opened in Webots.
 
+### Supported OSM Keywords
+
+ Only a subset of the OpenStreetMap keywords are handle by the importer. Here is the list of currently supported keywords per OpenStreetMap object type.
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]:
```suggestion
 Only a subset of the OpenStreetMap keywords are handled by the importer. Here is the list of keywords currently supported for each OpenStreetMap object type:
```

File: docs/automobile/openstreetmap-importer.md [hunk 2]
```diff
@@ -106,6 +103,96 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
...
+  - `leaf_type`: the value should be either `needleleaved` or `broadleaved`.
+  - `height`.
+  - `diameter_crown`.
+
+
+#### Way
+
+##### Building
+
+The [building](https://wiki.openstreetmap.org/wiki/Key:building) or [building:part](https://wiki.openstreetmap.org/wiki/Key:building:part) keywords will result in a 3 dimensional building. The following associated keywords are taken into account:
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]:
```suggestion
The [building](https://wiki.openstreetmap.org/wiki/Key:building) or [building:part](https://wiki.openstreetmap.org/wiki/Key:building:part) keyword will result in a 3 dimensional building. The following associated keywords are taken into account:
```

File: docs/automobile/openstreetmap-importer.md [hunk 3]
```diff
@@ -106,6 +103,96 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
...
+  - `building:material` or `material`: the value should be: `wood`, `old house`, `plaster`, `cement_block`, `concrete`, `glass` or `mirror`.
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]:
```suggestion
  - `building:material` or `material`: the value should be either: `wood`, `old house`, `plaster`, `cement_block`, `concrete`, `glass` or `mirror`.
```

File: docs/automobile/openstreetmap-importer.md [hunk 4]
```diff
@@ -106,6 +103,96 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
...
+  - `roof:material`: the value should be: `tile`, `roof_tiles`, `concrete`, `slate`, `asbestos` or `metal`.
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]:
```suggestion
  - `roof:material`: the value should be either: `tile`, `roof_tiles`, `concrete`, `slate`, `asbestos` or `metal`.
```

File: docs/automobile/openstreetmap-importer.md [hunk 5]
```diff
@@ -106,6 +103,96 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
...
+
+##### Road
+
+The [highway](https://wiki.openstreetmap.org/wiki/Key:highway) keyword will result in a road. The following associated keywords are taken into account:
+  - `name`.
+  - `lanes`.
+  - `layer`.
+  - `lanes:forward`.
+  - `lanes:backward`.
+  - `oneway`: `yes`.
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]: That line looks odd, should it be `oneway:yes` instead?

File: docs/automobile/openstreetmap-importer.md [hunk 6]
```diff
@@ -106,6 +103,96 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
...
+##### Area
+
+The following keywords will result in a flat area:
+  - `amenity`: if the value is `parking`.
+  - `landuse`.
+  - `waterway`.
+
+#### Relation
+
+The following keywords are taken into account, they work the exact same way as for [Way](#way):
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]:
```suggestion
The following keywords are taken into account, they work the exact same way as for [way](#way):
```

File: docs/automobile/openstreetmap-importer.md [hunk 7]
```diff
@@ -106,6 +103,96 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
...
+  - `parking`
+
+### Configuration File
+
+In addition to the defined keywords, a configuration file can be used to define the default value when a keyword is not defined (e.g. the default height of the buildings or of the buildings of a specific type) or to define which objects should or should not be imported.
+
+It is possible to set default values for all the entities of an object in the section determined by its name (e.g. `building` section for buildings, or `road` section for roads).
+And to set more specific default values for a specific type of the object in the section determined by its name followed by `_` followed by the value of the associated keyword (e.g. `road_primary` for roads whose `highway` keyword value is `primary`, or `building_residential` for buildings whose `building` keyword value is `residential`).
+
+A typical configuration file can be seen in [appendix](a-typical-openstreetmap-importer-configuration-file.md).
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]:
```suggestion
A typical configuration file is provided in [appendix](a-typical-openstreetmap-importer-configuration-file.md).
```

File: docs/automobile/openstreetmap-importer.md [hunk 8]
```diff
@@ -106,6 +103,96 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
...
+  - `natural`
+  - `landuse`
+  - `parking`
+
+### Configuration File
+
+In addition to the defined keywords, a configuration file can be used to define the default value when a keyword is not defined (e.g. the default height of the buildings or of the buildings of a specific type) or to define which objects should or should not be imported.
+
+It is possible to set default values for all the entities of an object in the section determined by its name (e.g. `building` section for buildings, or `road` section for roads).
+And to set more specific default values for a specific type of the object in the section determined by its name followed by `_` followed by the value of the associated keyword (e.g. `road_primary` for roads whose `highway` keyword value is `primary`, or `building_residential` for buildings whose `building` keyword value is `residential`).
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]: This is not a sentence (missing verb) and is hard to understand...

**Comment by DavidMansolino** [comment]
:+1:

File: docs/automobile/openstreetmap-importer.md [hunk 9]
```diff
@@ -106,6 +103,98 @@ Using JOSM you can easily edit a map in order to add some elements, correct some
...
+  - `parking`
+
+### Configuration File
+
+In addition to the defined keywords, a configuration file can be used to define the default value when a keyword is not defined (e.g. the default height of the buildings or of the buildings of a specific type) or to define which objects should or should not be imported.
+
+It is possible to set default values for all the entities of an object in the section determined by its name (e.g. `building` section for buildings, or `road` section for roads).
+
+
+It is also possible to set default values for a specific type of the object, the name of the section should respect the following syntax:`<<name of the object>>_<<value of the associated keyword>>` (e.g. `road_primary` for roads whose `highway` keyword value is `primary`, or `building_residential` for buildings whose `building` keyword value is `residential`).
```
Comment (omichel) [inline docs/automobile/openstreetmap-importer.md]:
```suggestion
It is also possible to set default values for a specific type of object, the name of the section should respect the following syntax: `<<name of the object>>_<<value of the associated keyword>>` (e.g. `road_primary` for roads whose `highway` keyword value is `primary`, or `building_residential` for buildings whose `building` keyword value is `residential`).
```

**Comment by DavidMansolino** [comment]
:+1:

**Comment by fabienrohrer** [comment]
👀 _(I'm checking the content.)_

**Comment by fabienrohrer** [comment]
All OSM fields seem to be documented, except the OSM `Road.maxspeed` which is also supported. It uses also the OSM `addr:country` field to compute the Webots `Road.speedLimit` following this table: http://wiki.openstreetmap.org/wiki/Speed_limits#Country_code.2Fcategory_conversion_table

**Comment by DavidMansolino** [comment]
Well spotted, I forgot this one as it is treated in a different way as the other (in a specific file). I will add it.

**Pull request merged**